src/Apps/Auction/Routes/__tests__/Utils/RegisterTestPage.tsx

```diff
@@ -0,0 +1,67 @@
+import { Checkbox } from "@artsy/palette"
+import { CountrySelect } from "Components/v2"
+import { expectOne, RootTestPage } from "DevTools/RootTestPage"
+
+export class RegisterTestPage extends RootTestPage {
+  get registerButton() {
+    return this.find("button").filterWhere(btn =>
+      btn.text().includes("Register")
+    )
+  }
+  get form() {
+    return expectOne(this.find("form"))
+  }
+  get nameInput() {
+    return expectOne(this.form.find(`input[name="name"]`))
+  }
+  get addressInput() {
+    return expectOne(this.form.find(`input[name="street"]`))
+  }
+  get cityInput() {
+    return expectOne(this.form.find(`input[name="city"]`))
+  }
+  get stateInput() {
+    return expectOne(this.form.find(`input[name="state"]`))
+  }
+  get postalCodeInput() {
+    return expectOne(this.form.find(`input[name="postalCode"]`))
+  }
+  get telephoneInput() {
+    return expectOne(this.form.find(`input[name="telephone"]`))
+  }
+  get countryInput() {
+    return expectOne(this.form.find(CountrySelect))
+  }
+  get agreeToTermsInput() {
+    return expectOne(this.form.find(Checkbox))
+  }
+  async fillFormWithValidValues() {
```

Comment:
19. We're not actually filling in values onto a page as we're not running a headless browser here. Instead we're invoking props with mocked change events.

Reply:
For reference it's pretty common to hear in the github [enzyme issues](https://github.com/airbnb/enzyme/issues/1999#issuecomment-460155725) that `wrapper.prop('onChange')({...whatever})` is preferable to `simulate`, and simulate is actually just an alias for the former and so becomes the source of bugs and confusion as it isn't a real DOM event.

Reply:
Yeah, I've heard that too. I think Cypress might be a good fit to actually run a browser through the flow, maybe via https://github.com/artsy/integrity.

Reply:
Offtopic, but wow these `continue` links!